src/java/org/jivesoftware/openfire/auth/JDBCAuthProvider.java

```diff
@@ -53,8 +63,35 @@
  * <li><tt>jdbcProvider.connectionString = jdbc:mysql://localhost/dbname?user=username&amp;password=secret</tt></li>
  * <li><tt>jdbcAuthProvider.passwordSQL = SELECT password FROM user_account WHERE username=?</tt></li>
  * <li><tt>jdbcAuthProvider.passwordType = plain</tt></li>
+ * <li><tt>jdbcAuthProvider.acceptPreHashedPassword = false</tt></li>
  * <li><tt>jdbcAuthProvider.allowUpdate = true</tt></li>
  * <li><tt>jdbcAuthProvider.setPasswordSQL = UPDATE user_account SET password=? WHERE username=?</tt></li>
+ * <li><tt>jdbcAuthProvider.bcrypt.cost = 12</tt></li>
+ * </ul>
+ *
+ * <p>External systems integrating with Openfire may only have access to a hashed password.  In this scenario, setting 
+ * jdbcAuthProvider.acceptPreHashedPassword = true will allow this AuthProvider to directly compare the input to 
+ * the hash stored in the database.  This configuration is the rough equivalent to allowing the scenario where EITHER 
+ * of the following authentication situations are allowed at the same time:<br>
+ * jdbcAuthProvider.passwordSQL = SELECT MD5(password) FROM user_account WHERE username=?<br>
+ * jdbcAuthProvider.passwordType = plain<br>
+ * -OR-<br>
+ * jdbcAuthProvider.passwordSQL = SELECT password FROM user_account WHERE username=?<br>
+ * jdbcAuthProvider.passwordType = md5<br></p>
+ * 
+ * <p>jdbcAuthProvider.passwordType can accept a comma separated string of password types.  This can be useful in 
+ * situations where legacy (ex/md5) password hashes were stored and then "upgraded" to a stronger hash algorithm.
+ * Hashes are executed left to right.</p>
+ * <p>Example Setting: "md5,sha1"<br>  
+ * Usage: password -><br>
+ * (md5)&nbsp;286755fad04869ca523320acce0dc6a4&nbsp;-><br>
+ * (sha1)&nbsp;0524b1fc84d315b08db890413e65260040b08caa&nbsp;-></p>

```

Comment:
Explanation of passwordType configuration (new, allowing comma separated - backwards compatible)